backend/app/treasury/tests/test_covalent.py

```diff
@@ -0,0 +1,327 @@
+            }
+        },
+    )
+
+    portfolio_hist_balances = await covalent._get_treasury_portfolio("", 1)
+
+    assert portfolio_hist_balances == {
+        "address": "0x6d6f636b65645f7472656173757279",
+        "quote_currency": "USD",
+        "chain_id": 1,
+        "items": covalent_portfolio_v2_transfers,
+    }
+
+
+@mark.asyncio
+async def test_get_treasury_portfolio_status_err(monkeypatch: MonkeyPatch):
+    monkeypatch.setattr(AsyncClient, "get", return_mocked_resp)
+    monkeypatch.setattr(MockResponse, "raise_for_status", raise_http_status_error_404)
+
+    with raises(HTTPStatusError) as error:
+        portfolio_hist_balances = await covalent._get_treasury_portfolio("", 1)
+
+        assert not portfolio_hist_balances
+        assert "mocked http status error" in error.value
+
+
+@mark.asyncio
+async def test_get_treasury_portfolio_bad_resp_json(monkeypatch: MonkeyPatch):
+    monkeypatch.setattr(AsyncClient, "get", return_mocked_resp)
+    monkeypatch.setattr(
+        MockResponse,
+        "json",
+        lambda *_: {"dtaa": "{...}"},
+    )
+
+    with raises(KeyError):
+        portfolio_hist_balances = await covalent._get_treasury_portfolio("", 1)
+
+        assert not portfolio_hist_balances
+
+
+@mark.asyncio
+async def test_get_treasury_portfolio_bad_json_err(monkeypatch: MonkeyPatch):
+    monkeypatch.setattr(AsyncClient, "get", return_mocked_resp)
+    monkeypatch.setattr(
+        MockResponse,
+        "json",
+        lambda *_: loads("bad json response"),
+    )
+
+    with raises(JSONDecodeError):
+        portfolio_hist_balances = await covalent._get_treasury_portfolio("", 1)
+
+        assert not portfolio_hist_balances
+
+
+@mark.asyncio
+async def test_get_treasury_success(monkeypatch: MonkeyPatch):
+    fake_provider = FakeRedis()
+    monkeypatch.setattr(AsyncClient, "get", return_mocked_resp)
+    monkeypatch.setattr("backend.app.treasury.adapters.covalent.db", fake_provider)
+
+    covalent_portfolio_v2_transfers.append(spam_token_transfer)
+    monkeypatch.setattr(
+        MockResponse,
+        "json",
+        lambda *_: {
+            "data": {
+                "address": "0x6d6f636b65645f7472656173757279",
+                "quote_currency": "USD",
+                "chain_id": 1,
+                "items": covalent_portfolio_v2_transfers,
+            }
+        },
+    )
+
+    treasury = await covalent.get_treasury(
+        await covalent.get_treasury_portfolio("", 1),
+        ["0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee"],
+    )
+
+    asset = covalent.ERC20(
+        token_name=covalent_portfolio_v2_transfers[0]["contract_name"],
+        token_symbol=covalent_portfolio_v2_transfers[0]["contract_ticker_symbol"],
+        token_address=covalent_portfolio_v2_transfers[0]["contract_address"],
+        balance=covalent_portfolio_v2_transfers[0]["holdings"][0]["close"]["quote"],
+    )
+    asset.risk_contribution = None
+    treasury.assets[0].risk_contribution = None
```

Comment:
That's not great that we have to change `treasury` before testing its structure, but that's ok for now (Treasury is going to be structurally simplified soon).